mips_api/__init__.py

```diff
@@ -125,7 +125,52 @@ def collect_chart(org_name, secrets):
     return mips_dict
 
 
-def list_tags(params, chart_dict, omit_list, extra_dict):
+def process_chart(chart_dict, omit_list, extra_dict):
+    '''
+    Process chart of accounts to remove unneeded programs,
+    and inject some extra (meta) programs.
+
+    5-digit codes are inactive and should be ignored.
+    8-digit codes are active, but only the first 6 digits are significant,
+    i.e. 12345601 and 12345602 should be deduplicated as 123456.
+    '''
+
+    # deduplicate on shortened numeric codes
+    # pre-populate with codes to omit to short-circuit their processing
+    found_codes = []
+    found_codes.extend(omit_list)
+
+    # output object
+    out_chart = {}
+
+    # inject our extra programs at the beginning
+    for code, name in extra_dict.items():
+        out_chart[code] = name
+
+    # add active short codes
+    for code, name in chart_dict.items():
+        if len(code) > 5: # only include active codes
+            short = code[:6]  # ignore the last two digits on active codes
+            if short not in found_codes:
+                out_chart[short] = name
+                found_codes.append(short)
+
+    return out_chart
+
+def filter_chart(params, raw_chart, omit_list, extra_dict):
+    '''
+    Optionally process the chart of accounts based on a query-string parameter."
+    '''
+
+    mips_dict = raw_chart
+
+    # if a 'filter' query-string parameter is defined, process the chart
```

Comment:
can we add something to the docs let the user know about this feature?  also maybe give them an idea on how to use it and what the result will be?